File: cs/Iris/ParserUtils/LexParseException/InvalidTypeOperand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Iris.ParserUtils.LexParseException
{
    public abstract class InvalidTypeOperand : BaseException
    {
        public Token.TokenValue tok;

        public InvalidTypeOperand(int lNum, int cNum, Token.TokenValue token)
            : base(lNum, cNum)
        {
            tok = token;
        }

        public string ExceptionToString(string side)
        {
            StringBuilder str = new StringBuilder();

            str.Append("invalid ");
            str.Append(side);
            str.Append(" operand for ");
            str.Append(Token.tokenstring(tok));

            return str.ToString();
        }
    }
}

File: java/Iris/src/com/rokuan/iris/exception/InvalidTypeOperand.java
package com.rokuan.iris.exception;

import com.rokuan.iris.parser.Token;

public abstract class InvalidTypeOperand extends BaseException
{
	public Token.TokenValue tok;

	public InvalidTypeOperand(int lNum, int cNum, Token.TokenValue token)
	{
		super(lNum, cNum);
		tok = token;
	}

	public String ExceptionToString(String side)
	{
		StringBuilder str = new StringBuilder();

		str.append("invalid ");
		str.append(side);
		str.append(" operand for ");
				str.append(Token.tokenstring(tok));

				return str.toString();
	}
}
